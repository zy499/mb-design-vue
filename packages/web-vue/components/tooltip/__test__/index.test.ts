import { nextTick } from 'vue';
import { mount } from '@vue/test-utils';
import Tooltip from '../index';

describe('Tooltip', () => {
  test('should render tooltip', async () => {
    const wrapper = mount(
      {
        template: [
          'top',
          'tl',
          'tr',
          'bottom',
          'bl',
          'br',
          'left',
          'lt',
          'lb',
          'right',
          'rt',
          'rb',
        ]
          .map(
            (item) =>
              `<a-tooltip content="content" :default-popup-visible="true" position="${item}" :render-to-body="false"><button>Button</button></a-tooltip>`
          )
          .join('/n'),
      },
      {
        global: {
          plugins: [Tooltip],
        },
      }
    );

    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should emit popupVisibleChange event', async () => {
    const wrapper = mount(Tooltip, {
      props: {
        mouseEnterDelay: 0,
        mouseLeaveDelay: 0,
      },
      slots: {
        default: '<button>Button</button>',
        content: 'Content',
      },
    });

    await wrapper.find('button').trigger('mouseenter');
    await wrapper.find('button').trigger('mouseleave');

    expect(wrapper.emitted('popupVisibleChange')).toHaveLength(2);
  });
});
