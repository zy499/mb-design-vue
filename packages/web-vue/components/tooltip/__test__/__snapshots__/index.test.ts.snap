// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should render tooltip 1`] = `
"<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-top mb-tooltip\\" style=\\"z-index: 1001;\\" trigger-placement=\\"top\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->/n<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-tl mb-tooltip\\" style=\\"z-index: 1002;\\" trigger-placement=\\"tl\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->/n<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-tr mb-tooltip\\" style=\\"z-index: 1003;\\" trigger-placement=\\"tr\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->/n<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-bottom mb-tooltip\\" style=\\"z-index: 1004;\\" trigger-placement=\\"bottom\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->/n<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-bl mb-tooltip\\" style=\\"z-index: 1005;\\" trigger-placement=\\"bl\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->/n<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-br mb-tooltip\\" style=\\"z-index: 1006;\\" trigger-placement=\\"br\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->/n<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-left mb-tooltip\\" style=\\"z-index: 1007;\\" trigger-placement=\\"left\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->/n<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-lt mb-tooltip\\" style=\\"z-index: 1008;\\" trigger-placement=\\"lt\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->/n<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-lb mb-tooltip\\" style=\\"z-index: 1009;\\" trigger-placement=\\"lb\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->/n<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-right mb-tooltip\\" style=\\"z-index: 1010;\\" trigger-placement=\\"right\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->/n<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-rt mb-tooltip\\" style=\\"z-index: 1011;\\" trigger-placement=\\"rt\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->/n<button>Button</button>
<!--teleport start-->
<div class=\\"mb-trigger-popup mb-trigger-position-rb mb-tooltip\\" style=\\"z-index: 1012;\\" trigger-placement=\\"rb\\" role=\\"tooltip\\">
  <transition-stub>
    <div class=\\"mb-trigger-popup-wrapper\\">
      <div class=\\"mb-trigger-content mb-tooltip-content\\">content</div>
      <div class=\\"mb-trigger-arrow mb-tooltip-popup-arrow\\"></div>
    </div>
  </transition-stub>
</div>
<!--teleport end-->"
`;
